store/tikv/client_batch.go

```diff
@@ -224,6 +248,7 @@ func (c *batchCommandsClient) reCreateStreamingClient(err error) bool {
 			zap.String("target", c.target),
 		)
 		c.client = streamClient
```

Comment:
this write becomes a write without hold "mutex", can other thread saw this change?

Reply:
`reCreateStreamingClient` acquire the write lock, so other send goroutine skips this `batchCommandsClient` and try the next one, in `tryLockForSend` @lysu

Reply:
`lockForRecreate` set a flag to exclusive others to step into this, but it call mutex.Unlock after set flag, so this function can exclusive other access, but I still confuse its visibility :confused:

Reply:
After `lockForRecreate`, the `reCreate` flag is set.  `tryLockForSend` will detect the flag.
Maybe you can implement a `RWMutex` struct with a `tryRlock` API, and then you'll figure out it.